core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -642,6 +643,29 @@ private[log] class Cleaner(val id: Int,
   }
 
   /**
+    * We want to get the last offset in the first log segment in segs.
+    * LogSegment.nextOffset() gives the exact last offset in a segment, but can be expensive since it requires
+    * scanning the segment from the last index entry.
+    * Therefore, we estimate the last offset of the first log segment by using
+    * the base offset of the next segment in the list.
+    * If the next segment doesn't exist, first Uncleanable Offset will be used.
+    *
+    * @param segs - remaining segments to group. The current head is segment were estimating the lastOffset for.
+    * @return The estimated last offset => The upper bound of the last offset
```

Comment:
Perhaps replacing "The estimated last offset => The upper bound of the last offset" with "The estimated last offset for the first segment in segs"?